### PR TITLE
Get visible pages and viewer configuration props

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**
Function:
- getVisiblePages
- setUrlExtraction
- setPageBorderVisibility
- setPageTransparencyGrid
- setDefaultPageColor
- setBackgroundColor

In which I am not quite sure about the effect of [urlExtraction](https://www.pdftron.com/api/ios/Classes/PTPDFViewCtrl.html#/c:objc(cs)PTPDFViewCtrl(im)SetUrlExtraction:) and could use some inputs. Everything else works fine. Thanks.

### CODE REVIEW

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 1]
```diff
@@ -2816,6 +2821,44 @@ public WritableMap getCanvasSize() {
...
+    public void setUrlExtraction(boolean urlExtraction) throws PDFNetException {
+        if (getPdfViewCtrl() != null) {
+            getPdfViewCtrl().setUrlExtraction(urlExtraction);
+        }
+    }
+
+    public void setPageBorderVisibility(boolean pageBorderVisibility) throws PDFNetException {
+        if (getPdfViewCtrl() != null) {
+            getPdfViewCtrl().setPageBorderVisibility(pageBorderVisibility);
+            getPdfViewCtrl().update();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2833]: update(true)

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -2816,6 +2821,44 @@ public WritableMap getCanvasSize() {
...
+        }
+    }
+
+    public void setPageTransparencyGrid(boolean pageTransparencyGrid) throws PDFNetException {
+        if (getPdfViewCtrl() != null) {
+            getPdfViewCtrl().setPageTransparencyGrid(pageTransparencyGrid);
+            getPdfViewCtrl().update();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2840]: update(true)

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: API.md [hunk 3]
```diff
@@ -1589,3 +1608,70 @@ this._viewer.getCanvasSize().then(({width, height}) => {
...
+
+Name | Type | Description
+--- | --- | ---
+backgroundColor | object | the background color, in the format `{red: number, green: number, blue: number}`, each number in range [0, 255]
+
+```js
+this._viewer.setBackgroundColor({red: 0, green: 0, blue: 255}); // blue color
+```
+
+#### defaultPageColor
```
Comment (sgong-pdftron) [inline API.md:1666]: setDefaultPageColor

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: ios/RNTPTDocumentView.m [hunk 4]
```diff
@@ -1873,6 +1872,60 @@ - (void)applyLayoutMode:(PTPDFViewCtrl *)pdfViewCtrl
...
+{
+    PTPDFViewCtrl *pdfViewCtrl = self.documentViewController.pdfViewCtrl;
+    [pdfViewCtrl SetPageTransparencyGrid:pageTransparencyGrid];
+    [pdfViewCtrl Update:YES];
+}
+
+- (void)setDefaultPageColor:(NSDictionary *)defaultPageColor
+{
+    if (defaultPageColor) {
+        NSArray *keyList = defaultPageColor.allKeys;
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:1897]:
```suggestion
        NSArray *keyList = defaultPageColor.allKeys;
        NSAssert([keyList containsObject:PTColorRedKey] &&
            [keyList containsObject:PTColorGreenKey] &&
            [keyList containsObject:PTColorBlueKey]);
```

File: ios/RNTPTDocumentView.m [hunk 5]
```diff
@@ -1873,6 +1872,60 @@ - (void)applyLayoutMode:(PTPDFViewCtrl *)pdfViewCtrl
...
+            
+            [pdfViewCtrl Update:YES];
+        }
+    }
+}
+
+- (void)setBackgroundColor:(NSDictionary *)backgroundColor
+{
+    if (backgroundColor) {
+        NSArray *keyList = backgroundColor.allKeys;
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:1915]:
```suggestion
        NSArray *keyList = backgroundColor.allKeys;
        NSAssert([keyList containsObject:PTColorRedKey] &&
            [keyList containsObject:PTColorGreenKey] &&
            [keyList containsObject:PTColorBlueKey]);
```

**Review by JamesTron — APPROVED** [review]

File: ios/RNTPTDocumentView.m (hunk 4 above)
Comment (Maxwell-Yang-PDFTron) [reply ios/RNTPTDocumentView.m:1965]: If I understand correctly, NSAssert would crash the app if condition is not true. So why is it better than an if check in this case?

File: ios/RNTPTDocumentView.m (hunk 5 above)
Comment (Maxwell-Yang-PDFTron) [reply ios/RNTPTDocumentView.m:1983]: Same as above.

File: ios/RNTPTDocumentView.m (hunk 4 above)
Comment (JamesTron) [reply ios/RNTPTDocumentView.m:1965]: Assert will "crash" the app, but only when debugging. Was I correct that not having one of these keys would constitute a programming error? If so, it is nice to alert the programmer in a way that's impossible to miss. But in release we can fail in a gentler way.

File: ios/RNTPTDocumentView.m (hunk 5 above)
Comment (JamesTron) [reply ios/RNTPTDocumentView.m:1983]: Also same as above.

**Review by JamesTron — APPROVED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Pull request merged**